app/script/event/EventRepository.js

```diff
@@ -648,7 +648,15 @@ z.event.EventRepository = class EventRepository {
    */
   _handleEventSaving(event, source) {
     const {conversation: conversationId, data: mappedData} = event;
-    const eventId = (mappedData && mappedData.replacing_message_id) || event.id;
+    /*
+     * We can detect an edit event if the event has a replacing_message_id and
+     * no preview yet.
+     * This way, we avoid collisions with link preview of edited messages which have
+     * replacing_message_id and a non empty array of previews.
+     */
+    const hasLinkPreview = mappedData && mappedData.previews && !!mappedData.previews.length;
+    const isEditEvent = mappedData && !hasLinkPreview && mappedData.replacing_message_id;
```

Comment:
A message can be both: A link preview and an edited message at the same time.

We need to handle all of these cases. With the change we also noticed that an edited message that is not a link preview should only ever replace an existing message. Message edits do not appear retroactively if the device did not exist at the time of the original message.